src/main/java/hudson/plugins/templateproject/ItemListenerImpl.java

```diff
@@ -47,11 +62,46 @@ public void onLoaded() {
 		}
 	}
 
+	@Override
+	public void onRenamed(Item item, String oldName, String newName) {
+		for (AbstractProject project : hudson.getAllItems(AbstractProject.class)) {
```

Comment:
This may cause a serious performance degradation on renames. But there is no other obvious way since we don't store backward links from templated projects.
I also doubt such rename hook will work correctly with folders. CC @jglick